### ISSUE TITLE
Janitor should clean up old finished flows

### ISSUE THREAD

**Issue by abador** [ory/hydra#3559]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/hydra/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/hydra/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Network](https://www.ory.sh/) project.
- [ ] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe your problem

At the moment janitor (in `FlushInactiveLoginConsentRequests`) cleans up the `hydra_oauth2_flow` for all states different than `flow.FlowStateConsentUsed`. This seems ok and cleans up any broken/unfinished flows. 
During our testing, we noticed that our database gets bloated and the data isn't flushed for short sessions/consents (combinations of  `consent_remember = 0|1, consent_remember_for = 0|60|3600, login_remember=0, login_remember_for=0 ` ).
If I understand correctly those records could also be deleted.
This also impacts `hydra_oauth2_code `and `hydra_oauth2_oidc `tables

### Describe your ideal solution

An additional cleanup job is added that cleans up flows that have the `state `column equal to `flow.FlowStateConsentUsed`, but aren't valid anymore (no other tables rely on them anymore). 

### Workarounds or alternatives

Running own query that cleans up the tables. This is doable if you have the same settings for every client, but I don't see a valid solution that would clean it up properly in other setups

### Version

2.1.1

**Comment by Renkas** [comment]
I'm also looking for a solution to clean up those tables.

**Comment by jderboven** [comment]
Also looking for an embedded solution for that

**Comment by UpenderOH** [comment]
even i am looking for the solution to cleanup the stale data from the tables as it drastically increases the table size on the production. Any one having query for it ?

**Comment by jderboven** [comment]
@UpenderOH as far as I understood, you have to be sure about the TTLs of the flow tokens as it seems this is the table getting super heavy.

What I did is those steps from now in a script..

    1. OAuth2 Authentication Sessions
       - Delete all sessions where ID is in the list of valid 'login_session_id'
    
    2. OAuth2 Access
       - Delete all records where 'challenge_id' is in recordsToDeleteIds list
    
    3. OAuth2 Codes
       - Delete all records where 'challenge_id' is in recordsToDeleteIds list
    
    4. OAuth2 OIDC
       - Delete all records where 'challenge_id' is in recordsToDeleteIds list
    
    5. OAuth2 PKCE (Proof Key for Code Exchange)
       - Delete all records where 'challenge_id' is in recordsToDeleteIds list
    
    6. OAuth2 Refresh Tokens
       - Delete all records where 'challenge_id' is in recordsToDeleteIds list

But there's also the logout_requests table getting flooded, I think we'd need some assistance for that

**Comment by UpenderOH** [comment]
> @UpenderOH as far as I understood, you have to be sure about the TTLs of the flow tokens as it seems this is the table getting super heavy.
> 
> What I did is those steps from now in a script..
> 
> ```
> 1. OAuth2 Authentication Sessions
>    - Delete all sessions where ID is in the list of valid 'login_session_id'
> 
> 2. OAuth2 Access
>    - Delete all records where 'challenge_id' is in recordsToDeleteIds list
> 
> 3. OAuth2 Codes
>    - Delete all records where 'challenge_id' is in recordsToDeleteIds list
> 
> 4. OAuth2 OIDC
>    - Delete all records where 'challenge_id' is in recordsToDeleteIds list
> 
> 5. OAuth2 PKCE (Proof Key for Code Exchange)
>    - Delete all records where 'challenge_id' is in recordsToDeleteIds list
> 
> 6. OAuth2 Refresh Tokens
>    - Delete all records where 'challenge_id' is in recordsToDeleteIds list
> ```
> 
> But there's also the logout_requests table getting flooded, I think we'd need some assistance for that

@jderboven thanks for the response. The above script you mention is with respect to new hydra v2 version database?

**Comment by jderboven** [comment]
@UpenderOH yes this is for V2, also I forgot to add a "AND active = false" on refresh table, it holds refresh tokens

**Comment by jderboven** [comment]
@UpenderOH have you found a solution ? The script is a fail for us so switching back to janitor but that ain't working

**Comment by SiebelsTim** [comment]
I think this is still important. There are tables that grow indefinitely. 
I am not sure how to tackle this without deeper understanding of the database tables or how to get this.
